File: Two Sum - Leetcode 1/Two Sum - Leetcode 1.js
var twoSum = function(nums, target) {
    const h = new Map();
    for (let i = 0; i < nums.length; i++) {
        h.set(nums[i], i);
    }

    for (let i = 0; i < nums.length; i++) {
        const y = target - nums[i];
        if (h.has(y) && h.get(y) !== i) {
            return [i, h.get(y)];
        }
    }

    throw new Error("No two sum solution");
};

File: Two Sum - Leetcode 1/Two Sum - Leetcode 1.java
import java.util.HashMap;
import java.util.Map;

public class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> h = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            h.put(nums[i], i);
        }

        for (int i = 0; i < nums.length; i++) {
            int y = target - nums[i];
            if (h.containsKey(y) && h.get(y) != i) {
                return new int[] {i, h.get(y)};
            }
        }

        throw new IllegalArgumentException("No two sum solution");
    }
}
